### ISSUE TITLE
Can't login github account

### ISSUE THREAD

**Issue by BailPlus** [microsoft/vscode#171587]
Does this issue occur when all extensions are disabled?: not applicable

- VS Code Version: 1.74.3 (installed via Deepin store)
- OS Version: Deepin GNU/Linux V20.8

Steps to Reproduce:

1. Login with my own github account.
2. Then it showed the following notice:
    ![image](https://user-images.githubusercontent.com/34512765/213108993-11763d3f-3f6f-4acc-8d5f-3b108fa1e949.png)
3. Retry again, and the problem still there.

The plugin outputs:
```
2023-01-18 15:23:08.918 [info] Getting sessions for read:user,repo,user:email,workflow...
2023-01-18 15:23:08.918 [info] Got 0 sessions for read:user,repo,user:email,workflow...
2023-01-18 15:23:08.933 [info] Getting sessions for read:user,repo,user:email...
2023-01-18 15:23:08.933 [info] Got 0 sessions for read:user,repo,user:email...
2023-01-18 15:23:08.950 [info] Getting sessions for read:user,repo,user:email,workflow...
2023-01-18 15:23:08.950 [info] Got 0 sessions for read:user,repo,user:email,workflow...
2023-01-18 15:23:10.170 [info] Logging in for the following scopes: read:user repo user:email workflow
2023-01-18 15:23:10.174 [info] Trying without local server... (read:user repo user:email workflow)
2023-01-18 15:23:17.346 [info] Exchanging code for token...
2023-01-18 15:23:18.139 [info] Token exchange success!
2023-01-18 15:23:18.141 [info] Getting user info...
2023-01-18 15:23:21.830 [error] Unexpected error parsing response from GitHub: invalid json response body at https://github.com/user reason: Unexpected token < in JSON at position 2
2023-01-18 15:23:21.831 [error] FetchError: invalid json response body at https://github.com/user reason: Unexpected token < in JSON at position 2
	at /opt/apps/com.visualstudio.code/files/code/resources/app/extensions/github-authentication/dist/extension.js:2:687068
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async t.GitHubServer.getUserInfo (/opt/apps/com.visualstudio.code/files/code/resources/app/extensions/github-authentication/dist/extension.js:2:1309674)
	at async t.GitHubAuthenticationProvider.o (/opt/apps/com.visualstudio.code/files/code/resources/app/extensions/github-authentication/dist/extension.js:2:1300184)
	at async t.GitHubAuthenticationProvider.createSession (/opt/apps/com.visualstudio.code/files/code/resources/app/extensions/github-authentication/dist/extension.js:2:1299667)
```
Then I opened https://github.com/user and discovered that there is already an account there. But it isn't mine.
What should I do?

**Comment by TylerLeonhardt** [comment]
This is very unusual because the error that that is logging:
https://github.com/microsoft/vscode/blob/8972dc89be6de79da7e96c7984c70887e618fe68/extensions/github-authentication/src/githubServer.ts#L534

Only happens when we run:
https://github.com/microsoft/vscode/blob/8972dc89be6de79da7e96c7984c70887e618fe68/extensions/github-authentication/src/githubServer.ts#L517
the important part being:
```
this.getServerUri('/user')
```
which is defined here:
https://github.com/microsoft/vscode/blob/8972dc89be6de79da7e96c7984c70887e618fe68/extensions/github-authentication/src/githubServer.ts#L504-L511

which would only return:
* `https://api.github.com/user`
* or if you were using GitHub Enterprise, and you set you baseurl to `https://github.com/` (which isn't correct, but let's say you did), that would be: `https://github.com/api/v3/user`

Neither of which you are seeing.

So my first suspicion is that this release of vscode isn't genuine because we don't ship to the deepin store. Can you tell me more about this package? Maybe a screenshot of the listing in the app store?

Additionally, can you send me the contents of:

```
/opt/apps/com.visualstudio.code/files/code/resources/app/extensions/github-authentication/dist/extension.js
```

**Comment by BailPlus** [comment]
[extension.js](https://github.com/microsoft/vscode/files/10462969/extension.js.txt)
Because github can't upload `.js` file, I changed the suffix to `.js.txt`. 
Additionally, I'll send the vscode package in deepin store to your email me@tylerleonhardt.com soon. 
Screenshot is there:
![image](https://user-images.githubusercontent.com/34512765/213613280-e302c9eb-093d-46ec-a544-2203d0f8d3ef.png)
You may need to translate it.

**Comment by BailPlus** [comment]
I knew where the bug was.
My computer couldn't connect to github recently. When I execute `ping github.com` on my computer, I got the IP address `127.0.0.1`. Then I searched the true IP address of github.com and added it into `/etc/hosts`. BUT  I set api.github.com the same IP of github.com by mistake. Recently, I fixed the IP address of api.github.com in `hosts`, and then I can login normally.
Thanks for developing this useful application and your patient attitude.
Today is the Spring Festival in China. Wish you a good luck in the new year!
